File: decorator/java/HotSauce.java
package decorator.java;

public class HotSauce extends Sauce {
    public HotSauce(Food food) {
        super(food);
    }

    @Override
    public void dip() {
        super.dip();
        System.out.println("Adding hot sauce to the food");
    }
    
}

File: decorator/golang/hotsauce.go
package main

type HotSauce struct {
	sauce Sauce
}

func NewHotSauce(sauce Sauce) *HotSauce {
	return &HotSauce{
		sauce,
	}
}

func (h *HotSauce) dip() {
	println("Adding hot sauce to the food and dipping it")
	h.sauce.dip()
}
